apps/institution-service/cypress/support/commands.ts

```diff
@@ -1,46 +1,86 @@
 /// <reference types="cypress" />
 import "@testing-library/cypress/add-commands";
+import { JwtPayload } from "jsonwebtoken";
 
-// ***********************************************
-// This example commands.ts shows you how to
-// create various custom commands and overwrite
-// existing commands.
-//
-// For more comprehensive examples of custom
-// commands please read more here:
-// https://on.cypress.io/custom-commands
-// ***********************************************
-//
-//
-// -- This is a parent command --
-// Cypress.Commands.add('login', (email, password) => { ... })
-//
-//
-// -- This is a child command --
-// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
-//
-//
-// -- This is a dual command --
-// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
-//
-//
-// -- This will overwrite an existing command --
-// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
-//
-// declare global {
-//   namespace Cypress {
-//     interface Chainable {
-//       login(email: string, password: string): Chainable<void>
-//       drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
-//     }
-//   }
-// }
+type ClientKeys = { clientId: string; clientSecret: string };
+
+type LoginArgs = {
+  storageKey: string;
+  grantType: "password" | "client_credentials";
+  audience: string;
+  clientId: string;
+  clientSecret: string;
+  scope?: string;
+  usernameEnvKey?: string;
+  passwordEnvKey?: string;
+};
+
+const login = (args: LoginArgs) => {
```

Comment:
Am I crazy or is half this function unused?

Reply:
I pulled this over from the auth service, which uses all of it. I'll pare it down since the institution service doesn't do any other grant-types.

Reply:
I refactored it to be simpler.

Reply:
I just saw your other comment. I'll move it into the main spec file so we don't have to mess with this.